python/pyarrow/_dataset.pyx

```diff
@@ -532,12 +532,12 @@ cdef class InMemoryDataset(Dataset):
 
     Parameters
     ----------
-    source : The data for this dataset.
+    source : The data for this dataset
```

Comment:
Why are the periods deleted? Other parameter descriptions, like the ones below, keep theirs.

Reply:
Actually this one makes sense; the description here is in the spot for the datatype, which isn't supposed to have a period. I think this comment needs to be fixed to have a type, then this description should go on the next line.

Reply:
Good eye. Numpydoc catches the first issue (the period) but doesn't catch the ultimate issue which is a poorly-formed docstring. I was focusing too narrowly while fixing these. I don't think we have an automated way of catching this type of issue (a non-type sitting in the position a type would normally go).